crates/sui-adapter/src/adapter.rs

```diff
@@ -562,95 +582,112 @@ fn handle_transfer<
     object_owner_map: &mut BTreeMap<SuiAddress, SuiAddress>,
     newly_generated_ids: &HashSet<ObjectID>,
 ) -> Result<(), ExecutionError> {
-    match type_ {
-        TypeTag::Struct(s_type) => {
-            let has_public_transfer = abilities.has_store();
-            // safe because `has_public_transfer` was properly determined from the abilities
-            let mut move_obj =
-                unsafe { MoveObject::new_from_execution(s_type, has_public_transfer, contents) };
-            let old_object = by_value_objects.remove(&move_obj.id());
-
-            #[cfg(debug_assertions)]
-            {
-                check_transferred_object_invariants(&move_obj, &old_object)
+    let s_type = match type_ {
```

Comment:
This isn't really that big of a change. Just some code styling changes to make it easier to read.